src/test/kotlin/uk/gov/justice/digital/hmpps/breachnoticeapi/integration/BreachNoticeCrudTests.kt

```diff
@@ -0,0 +1,230 @@
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+class BreachNoticeCrudTests : IntegrationTestBase() {
+
+  @Autowired
+  private lateinit var breachNoticeRepository: BreachNoticeRepository
+
+  @Test
+  fun `should create a breach notice`() {
+    webTestClient.post()
+      .uri("/breach-notice")
+      .headers(setAuthorisation(roles = listOf("ROLE_TEMPLATE_KOTLIN__UI")))
+      .bodyValue(
+        BreachNotice(
+          crn = "X00000B",
+        ),
+      )
+      .exchange()
+      .expectStatus()
+      .isCreated
+
+    var breachNotice: MutableList<BreachNoticeEntity> = breachNoticeRepository.findByCrn("X00000B")
+    assertThat(breachNotice.first().crn).isEqualTo("X00000B")
+    assertThat(breachNotice.first().id).isNotNull()
+  }
+
+  @Test
+  fun `should update a breach notice`() {
+    webTestClient.post()
+      .uri("/breach-notice")
+      .headers(setAuthorisation(roles = listOf("ROLE_TEMPLATE_KOTLIN__UI")))
+      .bodyValue(
+        BreachNotice(
+          crn = "X00001B",
+        ),
+      )
+      .exchange()
+      .expectStatus()
+      .isCreated
+
+    var breachNotice: MutableList<BreachNoticeEntity> = breachNoticeRepository.findByCrn("X00001B")
+    assertThat(breachNotice.first().crn).isEqualTo("X00001B")
+
+    webTestClient.put()
+      .uri("/breach-notice/" + breachNotice.first().id)
+      .headers(setAuthorisation(roles = listOf("ROLE_TEMPLATE_KOTLIN__UI")))
+      .bodyValue(
+        /* body = */
+        BreachNotice(
+          crn = "X00001B",
+          breachConditionTypeCode = "TYPE_CODE",
+          titleAndFullName = "Mr Joe Bloggs",
+          dateOfLetter = LocalDate.now(),
+          referenceNumber = "REFERENCE_NUMBER",
+          responseRequiredDate = LocalDate.now(),
+          breachNoticeTypeCode = "BRCH",
+          responsibleOfficer = "John Doe",
+          contactNumber = "01912525252",
+          nextAppointmentType = "NXTTYP",
+          nextAppointmentDate = LocalDateTime.now(),
+          nextAppointmentLocation = "NXT_LOCATION",
+          nextAppointmentOfficer = "APPT_OFFICER",
+          nextAppointmentContact = null,
+          completedDate = LocalDateTime.now(),
+          offenderAddress = Address(
+            addressId = 25,
+            type = "ENDO",
+            buildingName = "MOO",
+          ),
+          replyAddress = null,
+          basicDetailsSaved = true,
+        ),
+      )
+      .exchange()
+      .expectStatus()
+      .isOk
+
+    var updatedBreachNotice: MutableList<BreachNoticeEntity> = breachNoticeRepository.findByCrn("X00001B")
+    assertThat(updatedBreachNotice.first().crn).isEqualTo("X00001B")
+    assertThat(updatedBreachNotice.first().nextAppointmentLocation).isEqualTo("NXT_LOCATION")
+    assertThat(updatedBreachNotice.first().responsibleOfficer).isEqualTo("John Doe")
+    assertThat(updatedBreachNotice.first().basicDetailsSaved).isEqualTo(true)
```

Comment:
Because you're using data classes, I think you could just assert against the whole object in this test?

```suggestion
    val breachNoticeBody = BreachNotice(...
    
    webTestClient.put()
      .uri("/breach-notice/" + breachNotice.first().id)
      .headers(setAuthorisation(roles = listOf("ROLE_TEMPLATE_KOTLIN__UI")))
      .bodyValue(breachNoticeBody)
      ...
      
    val updatedBreachNotice = breachNoticeRepository.findByCrn("X00001B").single()
    assertThat(updatedBreachNotice.first().crn).isEqualTo("X00001B")
    assertThat(updatedBreachNotice).isEqualTo(breachNoticeBody)
```